Tests/GPU/CuptiTest/Exec/CUDA/mykernel.H

```diff
@@ -0,0 +1,32 @@
+#ifndef MY_KERNEL_H_
+#define MY_KERNEL_H_
+
+#include <AMReX_FArrayBox.H>
+
+AMREX_GPU_DEVICE AMREX_FORCE_INLINE
+void init_mf (int i, int j, int k,
+               amrex::Array4<amrex::Real> const& mf)
+{
+  mf(i,j,k) = 0.;
+}
+
+using clock_value_t = long long;
+
+__device__ void d_sleep(clock_value_t sleep_cycles)
+{
+  clock_value_t start = clock64();
+  clock_value_t cycles_elapsed;
+  do { cycles_elapsed = clock64() - start; } 
+  while (cycles_elapsed < sleep_cycles);
+}
+
+__global__
+void randomSleep ()
+{
+  // Sleep for random number of cycles
+  int cycles;
+  cycles = (int) (100000*amrex::Random());
+  d_sleep(cycles);
+}
+
```

Comment:
Should also make a device version of the sleep function and test this capturing an inlined lambda in AMReX fashion, so make sure it doesn't change the output or accuracy. (And with multiple nested device functions.)

Reply:
Added a test for inlined lambda launch of device sleep, no obvious changes to accuracy/stability. 
 Added also a test with a recursive device sleep.  calling, deviceSleep(n_cycles) or nestedDeviceSleep(n_cycles/n_levels, n_levels), i.e. recursive sleep over n_levels levels, sleeping for n_cycles/n_levels cycles per level, gives comparable results.